x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/20_semantic_text_field_mapper.yml

```diff
@@ -0,0 +1,153 @@
+setup:
+  - skip:
+      version: " - 8.12.99"
+      reason: semantic_text introduced in 8.13.0 # TODO change when 8.13.0 is released
+
+  - do:
+      inference.put_model:
+        task_type: sparse_embedding
+        inference_id: sparse-inference-id
+        body: >
+          {
+            "service": "test_service",
+            "service_settings": {
+              "model": "my_model",
+              "api_key": "abc64"
+            },
+            "task_settings": {
+            }
+          }
+  - do:
+      inference.put_model:
+        task_type: text_embedding
+        inference_id: dense-inference-id
+        body: >
+          {
+            "service": "text_embedding_test_service",
+            "service_settings": {
+              "model": "my_model",
+              "dimensions": 10,
+              "api_key": "abc64"
+            },
+            "task_settings": {
+            }
+          }
+
+  - do:
+      indices.create:
+        index: test-index
+        body:
+          mappings:
+            properties:
+              sparse_field:
+                type: semantic_text
+                model_id: sparse-inference-id
+              dense_field:
+                type: semantic_text
+                model_id: dense-inference-id
+              non_inference_field:
+                type: text
+
+---
+"Sparse vector results format":
+    - do:
+        index:
+          index: test-index
+          id: doc_1
+          body:
+            non_inference_field: "you know, for testing"
+            _semantic_text_inference:
+              sparse_field:
+                model_settings:
+                  inference_id: sparse-inference-id
+                  task_type: sparse_embedding
+                inference_results:
+                  - text: "inference test"
+                    inference:
+                      feature_1: 0.1
+                      feature_2: 0.2
+                      feature_3: 0.3
+                      feature_4: 0.4
+                  - text: "another inference test"
+                    inference:
+                      feature_1: 0.1
+                      feature_2: 0.2
+                      feature_3: 0.3
+                      feature_4: 0.4
+
+---
+"Dense vector results format":
+    - do:
+        index:
+          index: test-index
+          id: doc_1
+          body:
+            non_inference_field: "you know, for testing"
+            _semantic_text_inference:
+              dense_field:
+                model_settings:
+                  inference_id: sparse-inference-id
+                  task_type: text_embedding
+                  dimensions: 5
+                  similarity: cosine
+                inference_results:
+                  - text: "inference test"
+                    inference: [0.1, 0.2, 0.3, 0.4, 0.5]
+                  - text: "another inference test"
+                    inference: [-0.1, -0.2, -0.3, -0.4, -0.5]
```

Comment:
We should check that the mapping is as expected?

Reply:
We won't be doing mapping updates, just the raw indexing into the Lucene doc. How can we check that?

Reply:
I've added some tests for checking the mapping does not create additional fields, is that what you were thinking of?

Reply:
I was thinking of checking that the field is updated with the right parameters (dimensions and similarity).

Reply:
It is not - there is no change to the mappings that reflect this change:
-  The semantic_text field keeps having just `inference_id` and `type`.
- There are no other fields created, as the metadata field mapper deals with the results without creating mapping changes.

Is there any mapping change you'd expect from this operation?

Reply:
oh I see, the mapping change is on the inner field and they are not exposed in the mapping?

Reply:
That is correct. There is no mapping change, the metadata field mapper performs the indexing directly, without a mapping change.

AFAIU we can't do updates in the metadata field mapper for holding the changes. And we can't create a mapping with the same field path as a metadata field.

Is there something I'm missing?

Reply:
There's some unit tests provided by @Mikep86 that check the Lucene doc structure.

Also, we can check the end-to-end scenario once we have the semantic_query ready, so we can check retrieving docs work